src/store/fs/util/entity_manager.rs

```diff
@@ -0,0 +1,1322 @@
+        /// Channel to send internal commands to ourselves, to hand out to entity actors.
+        internal_send: mpsc::Sender<InternalCommand<P>>,
+        /// Map of live entity actors.
+        live: HashMap<P::EntityId, EntityHandle<P>>,
+        /// Global state shared across all entity actors.
+        state: P::GlobalState,
+        /// Pool of inactive entity actors to reuse.
+        pool: Vec<(
+            mpsc::Sender<entity_actor::Command<P>>,
+            entity_actor::Actor<P>,
+        )>,
+        /// Maximum size of the inbox of an entity actor.
+        entity_inbox_size: usize,
+        /// Initial capacity of the futures set for entity actors.
+        entity_futures_initial_capacity: usize,
+    }
+
+    impl<P: Params> ActorState<P> {
+        pub fn new(
+            state: P::GlobalState,
+            pool_capacity: usize,
+            entity_inbox_size: usize,
+            entity_response_inbox_size: usize,
+            entity_futures_initial_capacity: usize,
+        ) -> Self {
+            let (internal_send, internal_recv) = mpsc::channel(entity_response_inbox_size);
+            Self {
+                internal_recv,
+                internal_send,
+                live: HashMap::new(),
+                state,
+                pool: Vec::with_capacity(pool_capacity),
+                entity_inbox_size,
+                entity_futures_initial_capacity,
+            }
+        }
+
+        #[must_use = "this function may return a future that must be spawned by the caller"]
+        /// Friendly version of `spawn_boxed` that does the boxing
+        pub async fn spawn<F, Fut>(
+            &mut self,
+            id: P::EntityId,
+            f: F,
+        ) -> Option<impl Future<Output = ()> + Send + 'static>
+        where
+            F: FnOnce(SpawnArg<P>) -> Fut + Send + 'static,
+            Fut: Future<Output = ()> + Send + 'static,
+        {
+            self.spawn_boxed(
+                id,
+                Box::new(|x| {
+                    Box::pin(async move {
+                        f(x).await;
+                    })
+                }),
+            )
+            .await
+        }
+
+        #[must_use = "this function may return a future that must be spawned by the caller"]
+        pub async fn spawn_boxed(
+            &mut self,
+            id: P::EntityId,
+            f: Box<dyn FnOnce(SpawnArg<P>) -> future::Boxed<()> + Send>,
+        ) -> Option<impl Future<Output = ()> + Send + 'static> {
+            let (entity_handle, task) = self.get_or_create(id.clone());
+            let sender = entity_handle.send();
+            if let Err(e) =
+                sender.try_send(entity_actor::Command::Spawn(Spawn { id: id.clone(), f }))
+            {
+                match e {
+                    mpsc::error::TrySendError::Full(cmd) => {
+                        let entity_actor::Command::Spawn(spawn) = cmd else {
+                            panic!()
```

Comment:
Let's at least add an error message :)

Reply:
I guess it should be unreachable!, since we are calling this with Spawn a few lines further up.